modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/OAuth2RestClient.java

```diff
@@ -190,6 +193,35 @@ public List<ApplicationListItem> getApplicationsByClientId(String clientId) thro
         }
     }
 
+    /**
+     * Get application id using application name in an organization.
+     *
+     * @param appName          Application name.
+     * @param switchedM2MToken Switched m2m token generated for the given organization.
+     * @return Application id.
+     * @throws IOException If an error occurred while retrieving the application.
+     */
+    public String getAppIdUsingAppNameInOrganization(String appName, String switchedM2MToken) throws IOException {
+
+        String endPointUrl = subOrgApplicationManagementApiBasePath + "?filter=name eq " + appName;
+        endPointUrl = endPointUrl.replace(" ", "%20");
+
+        try (CloseableHttpResponse response = getResponseOfHttpGet(endPointUrl,
+                getHeadersWithBearerToken(switchedM2MToken))) {
+            String responseBody = EntityUtils.toString(response.getEntity());
+
+            ObjectMapper jsonWriter = new ObjectMapper(new JsonFactory());
+            ApplicationListResponse applicationResponse =
+                    jsonWriter.readValue(responseBody, ApplicationListResponse.class);
+            List<ApplicationListItem> applications = applicationResponse.getApplications();
+            if (applications != null && !applications.isEmpty()) {
+                return applications.get(0).getId();
+            } else {
+                return StringUtils.EMPTY;
+            }
```

Comment:
```suggestion
            }
            return StringUtils.EMPTY;
```

Reply:
Resolved with https://github.com/wso2/product-is/pull/20772/commits/1258e75f5c9e9525d851fd5e4143513bfc72946c